src/simplify_algebra.cpp

```diff
@@ -1065,6 +1065,13 @@ struct find_split_reshape
             return;
         }
 
+        if(std::any_of(split_outputs.begin(), split_outputs.end(), [](auto i) {
+               return i->outputs().front()->outputs().size() != 1;
```

Comment:
As it's written right now, it's kind of confusing. The `std::transform` in L1076 helps explain better what these variables are supposed to represent (and also some of the asserts shouldn't be needed anymore with the addition you made). `i->outputs().front()` is a contiguous op. You may want to explicitly make a variable `cont` like in L1078. Also, we have an assert for `i->outputs().size() == 1` in L1077. I'm not sure if you want to make the function return where the size is != 1 and remove that assert as well.

Reply:
Updated to remove both asserts and also add a comment to make the intent behind the check more clear